src/NodaTime/Duration.cs

```diff
@@ -1103,6 +1108,28 @@ public static Duration FromNanoseconds(BigInteger nanoseconds)
             return new Duration(days, nanoOfDay, noValidation: true);
         }
 
+#if NET7_0_OR_GREATER
+        /// <summary>
+        /// Converts a number of nanoseconds expressed as an <see cref="Int128"/> into a duration.
+        /// </summary>
+        /// <param name="nanoseconds">The number of nanoseconds to represent.</param>
+        /// <returns>A duration with the given number of nanoseconds.</returns>
+        public static Duration FromNanoseconds(Int128 nanoseconds)
+        {
+            if (nanoseconds < MinInt128Nanoseconds || nanoseconds > MaxInt128Nanoseconds)
+            {
+                throw new ArgumentOutOfRangeException(nameof(nanoseconds), $"Value should be in range [{MinNanoseconds}-{MaxNanoseconds}]");
```

Comment:
We probably should have a trivial test that values outside the range throw the expected exception/message, perhaps once for all of the `FromNanoseconds()` overloads?

Reply:
Yup, sounds good to me.

Reply:
It turns out we have FactoryMethods_OutOfRange already - I just need to add to that.